File: app/src/main/java/com/boykinchoi/star/design_pattern/adapter/animal/Cat.java
package com.boykinchoi.star.design_pattern.adapter.animal;

/**
 * @Author: 蔡佰健
 * @Description:
 * @Date:Create：in 2022/6/30 10:08
 */
public class Cat implements IAnimal{
    @Override
    public void makeSound() {
        System.out.println("猫：喵喵喵...");
    }
}

File: app/src/main/java/com/boykinchoi/star/generic/Cat.kt
package com.boykinchoi.star.generic

/**
 * @Author: 蔡佰健
 * @Description:
 * @Date:Create：in  2022/7/25 16:39
 */
class Cat : Animal() {

    fun clean() {
        println("cat clean")
    }

    override fun feed(){
        println("cat feed")
    }

}